src/Controller/Account/AccountController.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the user bundle package.
+ * (c) Connect Holland.
+ */
+
+namespace ConnectHolland\UserBundle\Controller\Account;
+
+use ConnectHolland\UserBundle\Entity\User;
+use ConnectHolland\UserBundle\Entity\UserInterface;
+use ConnectHolland\UserBundle\Event\UpdateEvent;
+use ConnectHolland\UserBundle\Event\UsernameUpdatedEvent;
+use ConnectHolland\UserBundle\Form\Account\AccountType;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\IsGranted;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
+use Twig\Environment;
+
+/**
+ * @codeCoverageIgnore WIP
+ * @Route({"en": "/account", "nl": "/account"}, name="connectholland_user_account")
+ */
+final class AccountController extends AbstractController
+{
+    /**
+     * @var UserPasswordEncoderInterface
+     */
+    private $encoder;
+
+    /**
+     * @var Environment
+     */
+    private $twig;
+
+    /**
+     * @var EventDispatcherInterface
+     */
+    private $eventDispatcher;
+
+    public function __construct(UserPasswordEncoderInterface $encoder, EventDispatcherInterface $eventDispatcher, Environment $twig)
+    {
+        $this->encoder         = $encoder;
+        $this->eventDispatcher = $eventDispatcher;
+        $this->twig            = $twig;
+    }
+
+    /**
+     * @Route({"en": "/details", "nl": "/gegevens"}, name="_account", methods={"GET", "POST"})
+     * @IsGranted("IS_AUTHENTICATED_FULLY")
+     */
+    public function edit(Request $request, FormFactoryInterface $formFactory): Response
+    {
+        /** @var UserInterface $user */
+        $user  = $this->getUser();
+        $email = $user->getEmail();
+        $form  = $formFactory->create(AccountType::class);
+        $form->setData($user);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $plainPassword = $form->get('plainPassword')->getData();
+
+            if ($email !== $user->getEmail()) {
+                $user->setPasswordRequestToken(bin2hex(random_bytes(32)));
```

Comment:
Perhaps this should be extracted in a future PR. A lot of places where this line is repeated

Reply:
Yes, especially because there is a todo for making a time based request token